server/app/vfg/parser/Problem_parser.py

```diff
@@ -100,6 +100,43 @@ def get_state_list(predicates_pattern_dic, text_block):
     return result
 
 
+def get_separate_state_list(predicates_pattern_dic, text_block):
+    """
+    This function turn all the predicate in text block into dictionary format 
+    and separate them into two list according to their predicate params
+    :param predicates_pattern_dic: dictionary which contain predicate name and predicate pattern pair
+    :param text_block: a text block contain list of predicates in plain text format
+    :return add_result: predicate lists in dictionary format that will be added
+    :return remove_result: predicate lists in dictionary format that will be removed
+    """
+    add_result = []
+    remove_result = []
+
+    for predicate_name, predicate_pattern in predicates_pattern_dic.items():
+        pattern = "((not\\s+)?\\(" + predicate_pattern + ")"
+        temp_pattern = re.compile(pattern)
+        predicates = temp_pattern.findall(text_block)
+        if predicates:
+            for predicate in predicates:
+                if predicate[1]:
+                    flag = False
+                    predicate = re.sub(r"not\s+", "", predicate[0])
+                else:
+                    flag = True
```

Comment:
rename `flag` -> `negated`, as it will convey more clearly the meaning of the conditional